language_service/src/tests.rs

```diff
@@ -0,0 +1,213 @@
+                        ),
+                    ),
+                ],
+            ),
+        ]
+    "#]],
+    );
+}
+
+#[test]
+fn package_type_update_causes_error() {
+    let errors = RefCell::new(Vec::new());
+    let mut ls = new_language_service(&errors);
+
+    ls.update_configuration(&WorkspaceConfigurationUpdate {
+        target_profile: None,
+        package_type: Some(PackageType::Lib),
+    });
+
+    ls.update_document("foo.qs", 1, "namespace Foo { operation Main() : Unit {} }");
+
+    expect_errors(
+        &errors,
+        &expect![[r#"
+        [
+            (
+                "foo.qs",
+                1,
+                [],
+            ),
+        ]
+    "#]],
+    );
+
+    ls.update_configuration(&WorkspaceConfigurationUpdate {
+        target_profile: None,
+        package_type: Some(PackageType::Exe),
+    });
+
+    expect_errors(
+        &errors,
+        &expect![[r#"
+            [
+                (
+                    "foo.qs",
+                    1,
+                    [
+                        Pass(
+                            EntryPoint(
+                                NotFound,
+                            ),
+                        ),
+                    ],
+                ),
+            ]
+        "#]],
+    );
+}
+
+#[test]
+fn target_profile_update_fixes_error() {
+    let errors = RefCell::new(Vec::new());
+    let mut ls = new_language_service(&errors);
+
+    ls.update_configuration(&WorkspaceConfigurationUpdate {
+        target_profile: Some(TargetProfile::Base),
+        package_type: Some(PackageType::Lib),
+    });
+
+    ls.update_document(
+        "foo.qs",
+        1,
+        r#"namespace Foo { operation Main() : Unit { if Zero == Zero { Message("hi") } } }"#,
+    );
+
+    expect_errors(
+        &errors,
+        &expect![[r#"
+    [
+        (
+            "foo.qs",
+            1,
+            [
+                Pass(
+                    BaseProfCk(
+                        ResultComparison(
+                            Span {
+                                lo: 45,
+                                hi: 57,
+                            },
+                        ),
+                    ),
+                ),
+                Pass(
+                    BaseProfCk(
+                        ResultLiteral(
+                            Span {
+                                lo: 45,
+                                hi: 49,
+                            },
+                        ),
+                    ),
+                ),
+                Pass(
+                    BaseProfCk(
+                        ResultLiteral(
+                            Span {
+                                lo: 53,
+                                hi: 57,
+                            },
+                        ),
+                    ),
+                ),
+            ],
+        ),
+    ]
+"#]],
+    );
+
+    ls.update_configuration(&WorkspaceConfigurationUpdate {
```

Comment:
Per other comment, it would be a good test to use a stdlib method that only available in Full (e.g. ResultAsBool), then switch to Base and see if you get an error for it.

Reply:
omg. So glad you asked me to add a test - I wasn't even passing in the `target_profile` to the stdlib compilation. Did you already notice that and were just subtly hinting at it??